strictdoc/core/file_traceability_index.py

```diff
@@ -496,10 +511,12 @@ def create_requirement(
                     )
                     if overwrite:
                         assert (
-                            requirement.reserved_uid
-                            not in function_name_to_reqs_uids
-                        )
-                    function_name_to_reqs_uids.append(requirement.reserved_uid)
+                            requirement.reserved_uid,
+                            file_reference.role,
```

Comment:
@stanislaw

> This will definitely cause merge conflicts but I can support you

Resolving conflicts was relatively easy. You still may want to double check I didn't mess up your intended semantics.

One particular place I noted is here. You were checking for duplicated UIDs in function_name_to_reqs_uids and class_name_to_reqs_uids. After my change, an entry is only considered duplicated if both UID and role match. So now it would be allowed to have to entries like `[("REQ1", None), ("REQ1", "TestCase")]` there.

Reply:
> One particular place I noted is here. You were checking for duplicated UIDs in function_name_to_reqs_uids and class_name_to_reqs_uids. After my change, an entry is only considered duplicated if both UID and role match. So now it would be allowed to have to entries like [("REQ1", None), ("REQ1", "TestCase")] there.

This makes sense.